### PR TITLE
change API with UI and add tooltip

### PR DESCRIPTION
**Pull request by gaoteng-git**
The corresponding UI PR: https://github.com/pytorch/kineto/pull/246
image
image
Pure CPU, without "GPU Metrics" panel:
image

### CODE REVIEW

File: tb_plugin/torch_tb_profiler/plugin.py [hunk 1]
```diff
@@ -151,6 +151,64 @@ def workers_route(self, request):
...
+                "Est. Achieved Occupancy:\n" \
+                "Occupancy is the ratio of active threads on an SM " \
+                "to the maximum number of active threads supported by the SM. " \
+                "The theoretical occupancy of a kernel is upper limit occupancy of this kernel, " \
+                "limited by multiple factors such as kernel shape, kernel used resource, " \
+                "and the GPU compute capability." \
+                "Est. Achieved Occupancy of a kernel, OCC_K = " \
+                "min(threads of the kernel / SM number / max threads per SM, theoretical occupancy of the kernel). " \
+                "This overall number is the weighted sum of all kernels OCC_K " \
+                "using kernel's execution duration as weight."
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/plugin.py:204]: It seems the method should belong to RunProfile class?

Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/plugin.py]: Done. Thanks.

File: tb_plugin/torch_tb_profiler/plugin.py [hunk 2]
```diff
@@ -162,15 +162,14 @@ def overview_route(self, request):
...
-            if profile.sm_efficency[gpu_id] > 0.0:
-                data["environments"].append({"title": "Est. SM Efficiency of GPU{}".format(gpu_id),
-                                             "value": "{} %".format(round(profile.sm_efficency[gpu_id] * 100, 2))})
-            if profile.occupancy[gpu_id] > 0.0:
-                data["environments"].append({"title": "Est. Achieved Occupancy of GPU{}".format(gpu_id),
-                                             "value": "{} %".format(round(profile.occupancy[gpu_id], 2))})
+        if len(profile.gpu_ids) > 0:
+            data["gpu_metrics"] = {}
+
+            gpu_metrics_data, has_occupancy, has_sm_efficiency = profile.get_gpu_metrics_data()
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/plugin.py:168]: could the get_gpu_metrics_data return tuple(data, tooltip) in one call?

Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/plugin.py]: Done. Thanks.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**